pkg/apis/awsprovider/v1alpha1/awsprovider.go

```diff
@@ -0,0 +1,39 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// AWSProviderSpec is the top level specification for the AWS Karpenter Provider.
+// This will contain configuration necessary to launch instances in AWS.
+type AWSProviderSpec struct {
+	// UserData to be applied to the provisioned nodes.
+	// It must be in the appropriate format based on the AMIFamily in use. Karpenter will merge certain fields into
+	// this UserData to ensure nodes are being provisioned with the correct configuration.
+	// +optional
+	UserData *string `json:"userData,omitempty"`
+}
+
+// AWSProvider is the Schema for the AWSProviders API
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=awsproviders,scope=Cluster,categories=karpenter
+type AWSProvider struct {
```

Comment:
Naming is hard. `kubectl get awsprovider` feels wrong to me, since it's not clearly scoped to anything about nodes. You could easily imagine a completely different CRD called AWSProvider for a different domain. Musing more on this, what do you think about calling it a `launchTemplate`? I guess this could conflict with an ACK `launchTemplate` CRD, but it's a better conceptual match.

Reply:
I'd lean towards `NodeConfig` or `AWSNodeConfig` or `AWSProvisionerConfig`, since a lot of the AWS specific instance config terminology is overloaded and confusing. LaunchTemplate or LaunchConfiguration are actual things which do not map to what our parameters are in the AWS CP.

Reply:
I like this. A couple other thoughts:
- AWSNodeConfiguration (fully spell it out, pretend configmaps are fully spelled :P )
- AWSNodeTemplate (this one has nice parity where NodeTemplate + AWSNodeTemplate make up a node)

Reply:
Hmm I chose Config because it's lesser to type :P 
With that I'd prefer `AWSNodeTemplate` than `AWSNodeConfiguration`. Agreed though that the work template might be better suited here. I'll make the change.